internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,491 @@
+		return nil
+	}
+	if kubeConfigSecret == nil {
+		r.Log.Errorf("Skipping creation of cluster deployment due to nil kubeconfigsecret")
+		return nil
+	}
+	if agentServiceConfig == nil {
+		r.Log.Errorf("Skipping creation of cluster deployment due to nil agentServiceConfig")
+		return nil
+	}
+	// Create a cluster deployment in the local cluster namespace
+	clusterDeployment := &hivev1.ClusterDeployment{
+		Spec: hivev1.ClusterDeploymentSpec{
+			Installed: true, // Let assisted know to import this cluster
+			ClusterMetadata: &hivev1.ClusterMetadata{
+				ClusterID:                "",
+				InfraID:                  "",
+				AdminKubeconfigSecretRef: v1.LocalObjectReference{Name: fmt.Sprintf("%s-admin-kubeconfig", r.LocalClusterName)},
+			},
+			ClusterInstallRef: &hivev1.ClusterInstallLocalReference{
+				Name:    r.LocalClusterName,
+				Group:   "extensions.hive.openshift.io",
+				Kind:    "AgentClusterInstall",
+				Version: "v1beta1",
+			},
+			Platform: hivev1.Platform{
+				AgentBareMetal: &agent.BareMetalPlatform{
+					AgentSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"infraenv": "local-cluster"},
+					},
+				},
+			},
+			PullSecretRef: &v1.LocalObjectReference{
+				Name: pullSecret.Name,
+			},
+		},
+	}
+	clusterDeployment.Name = r.LocalClusterName
+	clusterDeployment.Namespace = r.LocalClusterName
+	clusterDeployment.Spec.ClusterName = r.LocalClusterName
+	clusterDeployment.Spec.BaseDomain = dns.Spec.BaseDomain
+	agentClusterInstallOwnerRef := metav1.OwnerReference{
+		Kind:       "AgentServiceConfig",
+		APIVersion: "agent-install.openshift.io/v1beta1",
+		Name:       agentServiceConfig.Name,
+		UID:        agentServiceConfig.UID,
+	}
+	clusterDeployment.OwnerReferences = []metav1.OwnerReference{agentClusterInstallOwnerRef}
+	err := r.Client.Create(ctx, clusterDeployment)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		r.Log.Errorf("could not create ClusterDeployment due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) DeleteLocalClusterImportCRs(ctx context.Context) error {
+	err := r.DeleteClusterDeployment(ctx, r.LocalClusterName, r.LocalClusterName)
+	if err != nil {
+		r.Log.Errorf("could not delete local ClusterDeployment due to error %s", err.Error())
+		return err
+	}
+	err = r.DeleteAgentClusterInstall(ctx, r.LocalClusterName, r.LocalClusterName)
+	if err != nil {
+		r.Log.Errorf("could not delete local AgentClusterInstall due to error %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) ImportLocalCluster(ctx context.Context) error {
+
+	var errorList error
+
+	clusterVersion := &configv1.ClusterVersion{}
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      "version",
+	}
+	err := r.Client.Get(ctx, namespacedName, clusterVersion)
+	if err != nil {
+		errorList = multierror.Append(nil, errors.Wrap(err, "unable to find cluster version"))
```

Comment:
I'm not sure multierror is the best choice here.

Especially when you're creating a namespace and object in that namespace.